bin/utils.js

```diff
@@ -105,10 +106,44 @@ function bonjour (options) {
   });
 }
 
+function tryParseInt(input) {
+  const output = parseInt(input, 10);
+  if (Number.isNaN(output)) {
+    return null;
+  }
+  return output;
+}
+
+function findPort(server, defaultPort, defaultPortRetry, fn) {
+  let tryCount = 0;
```

Comment:
@u9520107 Just a simple NIT here, I think the code could be more readable in the following way, but feel free to not apply the changes!

```
function runPortFinder(tryCount, defaultPort, cb) {
  portfinder.basePort = defaultPort;
  tryCount += 1;
  portfinder.getPort((err, port) => {
    cb(err, port);
  });
}

function findPort(server, defaultPort, defaultPortRetry, fn) {
  let tryCount = 0;

  server.listeningApp.on('error', (err) => {
    if (err && err.code !== 'EADDRINUSE') {
      throw err;
    }

    if (tryCount >= defaultPortRetry) {
      fn(err);
      return;
    }

    runPortFinder(tryCount, defaultPort, fn);
  });

  runPortFinder(tryCount, defaultPort, fn);
}
```